tiled/client/awkward.py

```diff
@@ -0,0 +1,51 @@
+import awkward
+
+from ..serialization.awkward import from_zipped_buffers, to_zipped_buffers
+from ..structures.awkward import project_form
+from .base import BaseClient
+from .utils import handle_error
+
+
+class AwkwardArrayClient(BaseClient):
+    def __repr__(self):
+        # TODO Include some summary of the structure. Probably
+        # lift __repr__ code from awkward itself here.
+        return f"<{type(self).__name__}>"
+
+    def write(self, container):
+        handle_error(
+            self.context.http_client.put(
+                self.item["links"]["full"],
+                content=bytes(to_zipped_buffers(container, {})),
+                headers={"Content-Type": "application/zip"},
+            )
+        )
+
+    def read(self, slice=...):
+        structure = self.structure()
+        form = awkward.forms.from_dict(structure.form)
+        typetracer, report = awkward.typetracer.typetracer_with_report(
+            form,
+        )
+        proxy_array = awkward.Array(typetracer)
+        # TODO Ask awkward to promote _touch_data to a public method.
+        proxy_array[slice].layout._touch_data(recursive=True)
```

Comment:
This will become
```suggestion
        ak.typetracer.touch_data(proxy_array[slice])
```
once a new release has been made :)